datafusion/sql/src/expr/function.rs

```diff
@@ -66,8 +66,8 @@ pub fn suggest_valid_function(
 }
 
 /// Find the closest matching string to the target string in the candidates list, using edit distance(case insensitive)
-/// Input `candidates` must not be empty otherwise it will panic
-fn find_closest_match(candidates: Vec<String>, target: &str) -> String {
+/// Input `candidates` must not be empty otherwise an error is returned.
+fn find_closest_match(candidates: Vec<String>, target: &str) -> Result<String> {
```

Comment:
Wondering may be it can be an `Option<String>`?
Usually `find` methods, on collections as well returns the Option

Reply:
I tried this -- I'll put up a PR as a follow on.